### PR TITLE
[API/Devel] Retouch devel header

### PR DESCRIPTION
**Pull request by zhoonit**
- [API/Devel] Retouch devel header

```
**Changes proposed in this PR:**
- Change function sinature to exploit forward declaration
- s/export_to/exportTo
- Change header guard to be unique

**Self evaluation:**
1. Build test: [ ]Passed [ ]Failed [X]Skipped
2. Run test: [ ]Passed [ ]Failed [X]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

```

### CODE REVIEW

File: nntrainer/layers/layer_node.h [hunk 1]
```diff
@@ -255,7 +255,8 @@ class LayerNode : public ml::train::Layer, public GraphNode {
   virtual void
   export_to(Exporter &exporter,
             ExportMethods method = ExportMethods::METHOD_STRINGVECTOR) const {
-    exporter.saveResult(props, method, this);
+    /// waiting for #1272
+    // exporter.saveResult(props, method, this);
```
Comment (zhoonit) [inline nntrainer/layers/layer_node.h:259]: This was causing compile error (props does not exist, so this is commented) can be resolves with #1272

Comment (kparichay) [reply nntrainer/layers/layer_node.h:259]: please cherry-pick that commit on your local end. Once #1272 is merged to main, we can cherry pick it here too.

Comment (zhoonit) [reply nntrainer/layers/layer_node.h:259]: I'll after #1272 is merged

File: nntrainer/layers/layer_devel.h [hunk 2]
```diff
@@ -19,23 +19,27 @@
...
 #include <memory>
+#include <string>
 #include <vector>
 
-#include <layer_context.h>
-#include <node_exporter.h>
-
 namespace nntrainer {
 
+class InitContext;
```
Comment (kparichay) [inline nntrainer/layers/layer_devel.h:32]: good job.

File: nntrainer/layers/layer_devel.h [hunk 3]
```diff
@@ -153,15 +156,16 @@ class Layer {
...
-/**
- * @brief   Overriding output stream for layers and it's derived class
- */
-template <typename T, typename std::enable_if_t<
-                        std::is_base_of<Layer, T>::value, T> * = nullptr>
-std::ostream &operator<<(std::ostream &out, T &l) {
-  l.printPreset(out, Layer::PrintPreset::PRINT_SUMMARY);
-  return out;
-}
+/// @todo Decide where to put and how to implement(#987)
```
Comment (kparichay) [inline nntrainer/layers/layer_devel.h:159]: #987?

Comment (zhoonit) [reply nntrainer/layers/layer_devel.h:159]: lol it's #986,

**Review by kparichay — APPROVED** [review]
LGTM after rebase with #1272 merged main.

File: nntrainer/layers/layer_node.h (hunk 1 above)
Comment (zhoonit) [reply nntrainer/layers/layer_node.h:259]: We dicsussed offline to wait for #1284 instead.

**Review by lhs8928 — APPROVED** [review]
LGTM

**Review by myungjoo — APPROVED** [review]

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Pull request merged**